### PR TITLE
Fix acceptance tests - TCA and Prod

### PR DESCRIPTION
**Pull request by monishapadmavathi**
This PR contains testcase fixing 

- Export tests
- Integration tests
- Auth tests

Modifiy jenkins file to run both TCA and prod tests

### CODE REVIEW

File: genesyscloud/telephony_providers_edges_site/resource_genesyscloud_telephony_providers_edges_site_test.go [hunk 1]
```diff
@@ -19,6 +20,20 @@ import (
 	"github.com/mypurecloud/platform-client-sdk-go/v129/platformclientv2"
 )
 
+func assignregion() string {
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_site/resource_genesyscloud_telephony_providers_edges_site_test.go:23]: We will most likely need to use this function in a few tests, you can move it into `test_utils.go` and import it where needed

File: genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go [hunk 2]
```diff
@@ -57,6 +58,282 @@ type WrapupcodeExport struct {
...
+			{ResourceName: "test-queue-dev", Name: "test-queue-" + uuid.NewString() + "-dev", Description: "This is a dev queue", AcwTimeoutMs: 200000},
+		}
+
+		wrapupCodeResources = []WrapupcodeExport{
+			{ResourceName: "test-wrapupcode-prod", Name: "test-wrapupcode-" + uuid.NewString() + "-prod"},
+			{ResourceName: "test-wrapupcode-test", Name: "test-wrapupcode-" + uuid.NewString() + "-test"},
+			{ResourceName: "test-wrapupcode-dev", Name: "test-wrapupcode-" + uuid.NewString() + "-dev"},
+		}
+	)
+	//defer os.RemoveAll(exportTestDir)
```
Comment (dginty4) [inline genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go:214]: Is this needed?

File: genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go [hunk 3]
```diff
@@ -57,6 +58,282 @@ type WrapupcodeExport struct {
...
+			{ResourceName: "test-user-1", Name: "test-user-1", Email: "test-user-1" + uuid.NewString() + "@test.com" + uniquePostfix, State: "active"},
+			{ResourceName: "test-user-2", Name: "test-user-2", Email: "test-user-2" + uuid.NewString() + "@test.com" + uniquePostfix, State: "active"},
+		}
+
+		wrapupCodeResources = []WrapupcodeExport{
+			{ResourceName: "test-wrapupcode-1", Name: "test-wrapupcode-1-" + uuid.NewString() + uniquePostfix},
+			{ResourceName: "test-wrapupcode-2", Name: "test-wrapupcode-2-" + uuid.NewString() + uniquePostfix},
+		}
+	)
+	//defer os.RemoveAll(exportTestDir)
```
Comment (dginty4) [inline genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go:292]: Same with this?

**Review by dginty4 — APPROVED** [review]

File: genesyscloud/telephony_providers_edges_site/resource_genesyscloud_telephony_providers_edges_site_test.go (hunk 1 above)
Comment (monishapadmavathi) [reply genesyscloud/telephony_providers_edges_site/resource_genesyscloud_telephony_providers_edges_site_test.go]: Moved it to test_utils.go

File: genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go (hunk 3 above)
Comment (monishapadmavathi) [reply genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go]: Reverted it back.

File: genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go (hunk 2 above)
Comment (monishapadmavathi) [reply genesyscloud/tfexporter/resource_genesyscloud_tf_export_test.go]: Reverted it

**Review by dginty4 — APPROVED** [review]

File: genesyscloud/util/test_utils.go [hunk 4]
```diff
@@ -374,3 +374,18 @@ func RandString(length int) string {
...
+	)
+
+	if v := os.Getenv("GENESYSCLOUD_REGION"); v == "tca" {
+		region = "us-east-1"
+	} else if v == "us-east-1" {
+		region = "us-west-2"
+	}
+	regionJSON := "[" + strconv.Quote(region) + "]"
+	return regionJSON
+}
```
Comment (charliecon) [inline genesyscloud/util/test_utils.go:391]: If it is possible for the value in `GENESYSCLOUD_REGION` to not be "tca" or "us-east-1" , then this function will return `[""]` 

Should we maybe return a default region in this case?

**Review by charliecon — APPROVED** [review]

File: genesyscloud/util/test_utils.go (hunk 4 above)
Comment (monishapadmavathi) [reply genesyscloud/util/test_utils.go:389]: modified it to return default region

**Review by charliecon — APPROVED** [review]

**Pull request merged**